models/schedule_item.go

```diff
@@ -66,11 +67,76 @@ func NewTimeslotItem(t *myradio.Timeslot, finish time.Time, u func(*myradio.Time
 		Desc:    t.Description,
 		Start:   t.StartTime,
 		Finish:  finish,
-		Block:   "regular", // TODO(MattWindsor91): get this from elsewhere
+		Block:   getBlock(t.Title, t.StartTime),
 		PageURL: url.Path,
 	}, nil
 }
 
+func getBlock(name string, StartTime time.Time) string{
+	name = strings.ToLower(name)
+
+	type blockMatch struct {
+		nameFragment string
+		block string
+	}
+	var blockMatches = []blockMatch {
+		{"ury: early morning", "flagship"},
+		{"ury breakfast", "flagship"},
+		{"ury lunch", "flagship"},
+		{"ury brunch", "flagship"},
+		{"URY Brunch", "flagship"},
+		{"URY Afternoon Tea:", "flagship"},
+		{"URY:PM", "flagship"},
+		{"ury news", "news"},
+		{"ury sports", "news"},
+		{"ury football", "news"},
+		{"york sport report", "news"},
+		{"university radio talk", "news"},
+		{"candidate interview night", "news"},
+		{"election results night", "news"},
+		{"yusu election", "news"},
+		{"The Second Half With Josh Kerr", "news"},
+		{"URY SPORT", "news"},
+		{"ury speech", "speech"},
+		{"yorworld", "speech"},
+		{"in the stalls", "speech"},
+		{"screen", "speech"},
+		{"stage", "speech"},
+		{"game breaking", "speech"},
+		{"radio drama", "speech"},
+		{"Book Corner", "speech"},
+		{"Saturated Facts", "speech"},
+		{"URWatch", "speech"},
+		{"Society Challenge", "speech"},
+		{"Speech Showcase", "speech"},
+		{"roses live 201", "event"},
+		{"roses 201", "event"},
+		{"woodstock", "event"},
+		{"movember", "event"},
+		{"panto", "event"},
+		{"101:", "event"},
+		{"Vanbrugh Chair Debate", "event"},
+		{"URY Does RAG Courtyard Takeover", "event"},
+		{"URY Presents", "event"},
+		{"URYOnTour", "event"},
+		{"URY On Tour", "event"},
```

Comment:
Some of these shows don't exist anymore so it might be worth removing those

Reply:
Although they don’t exist would we not want to keep it for historical purposes?